src/core/ext/client_channel/client_channel.c

```diff
@@ -1096,6 +1099,9 @@ void grpc_client_channel_finish_initialization(
       chand->exit_idle_when_lb_policy_arrives) {
     chand->started_resolving = true;
     GRPC_CHANNEL_STACK_REF(chand->owning_stack, "resolver");
+    if (chand->lb_policy != NULL) {
```

Comment:
Good point. Removed. I was looking for obvious references to chand->lb_policy but missed the transitive one through the resolver.